ratis-server/src/main/java/org/apache/ratis/server/DataStreamServerFactory.java

```diff
@@ -36,4 +38,11 @@ static DataStreamServerFactory cast(DataStreamFactory dataStreamFactory) {
    * Server implementation for streaming in Raft group
    */
   DataStreamServerRpc newDataStreamServerRpc(RaftPeer server);
+
+  /**
+   * Server implementation for streaming in Raft group. The server will forward requests
+   * to peers.
+   */
+  DataStreamServerRpc newDataStreamServerRpc(
+      RaftPeer server, List<RaftPeer> peers, RaftProperties properties);
```

Comment:
I have added this new API that assume servers can have two modes:
1. with peers parameter this server forwards message to peers (if the server is initialized by `ataStreamServerRpc newDataStreamServerRpc(RaftPeer server, List<RaftPeer> peers, RaftProperties properties);`
2. without peers parameter this server does not forward messages (if the server is initialized by `DataStreamServerRpc newDataStreamServerRpc(RaftPeer server);`)